### PR TITLE
added CACHE_SIZE docs

### LINKED ISSUE #1324: Add `CACHE_SIZE` to docs

### PR DESCRIPTION
**Pull request by DvirDukhan**
closes #1324

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -36,6 +36,22 @@ $ redis-server --loadmodule ./redisgraph.so THREAD_COUNT 4
 
 ---
 
+## CACHE_SIZE
+
+The number of execution plans cached for each RedisGraph thread. Each thread in RedisGraph thread pool holds an execution plan cache where it stores execution plan templates for previously seen queries. When a new query and execution plan are needed to be store, and the cache is full, meaning the cache has reached the size of `CACHE_SIZE`, it will evict the least recently used (LRU) entry.
```
Comment (swilly22) [inline docs/configuration.md:41]: `The number of execution plans` don't go into details, from the user perspective we're caching queries.

Comment (swilly22) [inline docs/configuration.md:41]: `for each RedisGraph thread` once again too much details for the common user.
a user thinks there's a single cache per graph.

File: docs/configuration.md [hunk 2]
```diff
@@ -36,6 +36,22 @@ $ redis-server --loadmodule ./redisgraph.so THREAD_COUNT 4
...
+
```
Comment (swilly22) [inline docs/configuration.md:42]: Generally speaking this doc is too detailed and revels too much of the internals, please try to simplify it for the common user.

**Review by swilly22 — CHANGES_REQUESTED** [review]

**Review by swilly22 — DISMISSED** [review]
Much better, @jeffreylovitz please rephrase as needed.

File: docs/configuration.md [hunk 3]
```diff
@@ -36,6 +36,22 @@ $ redis-server --loadmodule ./redisgraph.so THREAD_COUNT 4
 
 ---
 
+## CACHE_SIZE
+
+The number of queries cached in RedisGraph. When a new query is needed to be store, and the cache is full, meaning the cache has reached the size of `CACHE_SIZE`, it will evict the least recently used (LRU) entry.
```
Comment (jeffreylovitz) [inline docs/configuration.md:41]: Suggest rewording the start of this line:
`The max number of queries for RedisGraph to cache. When a new query is encountered and the cache is full,`

**Review by jeffreylovitz — CHANGES_REQUESTED** [review]

**Review by swilly22 — APPROVED** [review]

**Review by swilly22 — APPROVED** [review]
